cdap-ui/app/cdap/components/DataPrepConnections/KafkaConnection/index.js

```diff
@@ -0,0 +1,402 @@
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { Component, PropTypes } from 'react';
+import { Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';
+import {objectQuery} from 'services/helpers';
+import NamespaceStore from 'services/NamespaceStore';
+import MyDataPrepApi from 'api/dataprep';
+import T from 'i18n-react';
+import LoadingSVG from 'components/LoadingSVG';
+import HostPortEditor from 'components/DataPrepConnections/KafkaConnection/HostPortEditor';
+import shortid from 'shortid';
+import ee from 'event-emitter';
+import CardActionFeedback from 'components/CardActionFeedback';
+
+const PREFIX = 'features.DataPrepConnections.AddConnections.Kafka';
+
+const LABEL_COL_CLASS = 'col-xs-3 col-form-label text-xs-right';
+const INPUT_COL_CLASS = 'col-xs-8';
+
+require('./KafkaConnection.scss');
+
+export default class KafkaConnection extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      name: '',
+      brokersList: [{
+        host: 'localhost',
+        port: '9092',
+        uniqueId: shortid.generate()
+      }],
+      connectionResult: null,
+      testConnectionLoading: false,
+      error: null,
+      loading: false
+    };
+
+    this.eventEmitter = ee(ee);
+    this.preventDefault = this.preventDefault.bind(this);
+    this.addConnection = this.addConnection.bind(this);
+    this.editConnection = this.editConnection.bind(this);
+    this.testConnection = this.testConnection.bind(this);
+    this.handleBrokersChange = this.handleBrokersChange.bind(this);
+  }
+
+  componentWillMount() {
+    if (this.props.mode === 'ADD') { return; }
+
+    this.setState({ loading: true });
+
+    let namespace = NamespaceStore.getState().selectedNamespace;
+
+    let params = {
+      namespace,
+      connectionId: this.props.connectionId
+    };
+
+    MyDataPrepApi.getConnection(params)
+      .subscribe((res) => {
+        let info = objectQuery(res, 'values', 0),
+            brokers = objectQuery(info, 'properties', 'brokers');
+
+        let name = this.props.mode === 'EDIT' ? info.name : '';
+        let brokersList = this.parseBrokers(brokers);
+
+        this.setState({
+          name,
+          brokersList,
+          loading: false
+        });
+      }, (err) => {
+        console.log('failed to fetch connection detail', err);
+
+        this.setState({
+          loading: false
+        });
+      });
+  }
+
+  parseBrokers(brokers) {
+    let brokersList = [];
+
+    brokers.split(',').forEach((broker) => {
+      let split = broker.trim().split(':');
```

Comment:
You don't have to do this but since this seems to be pattern that is used in a few places, you can shorten this to something like this,

```
let [host='', port=''] = broker.trim.split(':');
let obj = {host, port, uniqueId: shortid.generate()};
...
```